lti/views/lti_launches.py

```diff
@@ -1,16 +1,46 @@
 from pyramid.view import view_config
-from pyramid.httpexceptions import HTTPFound
 from lti.util.lti_launch import lti_launch
+from lti.util.view_renderer import view_renderer
+from lti.models.module_item_configuration import ModuleItemConfiguration
 
-@view_config(route_name='lti_launches', renderer='lti:templates/lti_launches/create_lti_launch.html.jinja2', request_method='POST')
+@view_config(route_name='lti_launches',  request_method='POST')
 @lti_launch
 def lti_launches(request):
   """
-    This is the main lti launch route. It renders the iframe that shows the via hypothesis app.
-    This will fail if hit from any other context.
+    This is the primary lti launch route. There are 3 views that could be rendered:
+      1. If a student launches before a teacher has configured the document then it will
+         display a message say that the teacher still needs to configure the document.
+
+      2. If a student or teach launch after the document has been configured then it diplays the
+         document with the annotation tools.
+
+      3. If a teacher launches and no document has been configured, it renders a form that allows
+         them to configure the document.
   """
+  if 'url' not in request.params:
+    config = request.db.query(ModuleItemConfiguration).filter(
+      ModuleItemConfiguration.resource_link_id == request.params['resource_link_id'] and
+      ModuleItemConfiguration.tool_consumer_instance_guid == request.params['tool_consumer_instance_guid']
+    )
+    if config.count() == 1:
```

Comment:
We need a way to deal with possibly more than one value being returned. 
You could add add a test for config.count() > 1 and throw an exception. 

Or use one() instead of filter(). You could catch the NoResultFound exception and do the "return _new_module_item_configuration". 

I'm don't think at this point in the code we have enough information to know what to do if more than 1 value is returned, so the exception would have to be passed up the call stack. We should think about where we would catch it and how to deal with it.

Reply:
If there is more than one then we have a more serious problem. That would indicate that either two lms's have the same unique identifier or two module items have the same unique identifier either way that is a big bug in whatever lms this would surface in. The way I have written that though will bury that problem so i will make it >= 1 so that an exception will get thrown. And the we can talk about a way to handle these types of errors.